src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { getNewTodoId, getUserById } from './components/helpers/helpers';
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(usersFromServer, todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setIsUserValid(true);
+    }
+
+    if (!title.trim()) {
+      setIsTitleValid(true);
+    }
+
+    if (!userId || !title.trim()) {
```

Comment:
you repeat `title.trim()` 3 times, let's create a separate variable for this.

Reply:
I've used it only twice. Making a seperate variable means adding the same amount of code

Reply:
Check 30, 34, 43 lines of your code. 

if you create a variable for `title.trim()` it means that you will apply trim method for your title once, it is kinda optimization

Reply:
right you are!